packages/block-library/src/pullquote/block.json

```diff
@@ -17,22 +17,24 @@
 			"default": "",
 			"__experimentalRole": "content"
 		},
-		"mainColor": {
-			"type": "string"
-		},
-		"customMainColor": {
-			"type": "string"
-		},
 		"textColor": {
 			"type": "string"
-		},
-		"customTextColor": {
-			"type": "string"
 		}
 	},
 	"supports": {
 		"anchor": true,
-		"align": [ "left", "right", "wide", "full" ]
+		"align": [ "left", "right", "wide", "full" ],
+		"color": {
+			"gradients": true,
+			"background": true,
+			"link": true
+		},
+		"__experimentalBorder": {
+			"color": true,
+			"radius": true,
+			"style": true,
+			"width": true
+		}
```

Comment:
We could simplify this config and simply make it `"__experimentalBorder": true`.

I don't see many new border support features being added on top of what's already there so it should be safe to do.

Reply:
Ah interesting. 

Out of curiosity, the properties don't matter then for these [specific checks](https://github.com/WordPress/gutenberg/blob/933c45a70808b590ca7c7a44cf4a73aafe53d35e/lib/block-supports/border.php#L17), e.g., `gutenberg_block_has_support( $block_type, array( '__experimentalBorder', 'radius' ) )`?

Reply:
Hmm, seems like they do matter though? 🤔 

https://github.com/WordPress/gutenberg/blob/933c45a70808b590ca7c7a44cf4a73aafe53d35e/lib/blocks.php#L388-L395

Reply:
> Hmm, seems like they do matter though? 🤔

That's what I can't work out 🤣 

I tested with `"__experimentalBorder": true` in pullquote/block.json and things were working, which is nice. 

But I couldn't see how they were working given the specific checks in blocks/border.php. 🤷

Reply:
Regardless, personally it has helped me having the support properties printed there explicitly, only as a convenient reminder of what the block supports :D

Reply:
> Hmm, seems like they do matter though?

This was an oversight of mine [when I refactored the border support](https://github.com/WordPress/gutenberg/pull/30124/commits/d87a2999b0cac30d479628b24ed4b9169a40fd93) to use the suggested `gutenberg_block_has_support`. Given this function doesn't check if the parent property is true it created a disjoint between support for static blocks and dynamic blocks.

I'm not certain if all block supports will wan't to allow the blanket opt-in so in place of updating `gutenberg_block_has_support` I've created a new PR to fix up the server side block support checks for border support, https://github.com/WordPress/gutenberg/pull/31020.

> That's what I can't work out 🤣
>
> I tested with "__experimentalBorder": true in pullquote/block.json and things were working, which is nice.

It "works" in this case because the only block support checks for the pullquote block are done via JS and the editor. The resulting CSS classes and styles from the block support are saved into the block's post content. This content is simply spat out on the frontend as is. 

A dynamic block such as the `core/post-tags` block is rendered via PHP. In this situation, the CSS classes and styles are retrieved via the code referenced in `blocks/border.php`

> Regardless, personally it has helped me having the support properties printed there explicitly, only as a convenient reminder of what the block supports :D

There isn't any real downside in being explicit here regarding which support is being opted into. So feel free to leave as is. 

If a block support feature such as `__experimentalBorder` is opted into via a boolean true value, it would mean that it would automatically adopt any new sub-features in the future. That could be a double edged sword depending on the scenario. Regarding borders, I don't think much will change there.

I'm glad the discussion came up though as it highlighted the disjoint between the editor and server side checks for support. Thanks! 👍